remote/remote_state_s3.go

```diff
@@ -193,23 +204,16 @@ func configValuesEqual(config map[string]interface{}, existingBackend *Terraform
 		}
 	}
 
-	// Nowadays it only makes sense to set the "dynamodb_table" attribute as it has
-	// been supported in Terraform since the release of version 0.10. The deprecated
-	// "lock_table" attribute is either set to NULL in the state file or missing
-	// from it altogether. Display a deprecation warning when the "lock_table"
-	// attribute is being used.
-	if util.KindOf(config["lock_table"]) == reflect.String && config["lock_table"] != "" {
-		terragruntOptions.Logger.Printf("%s\n", lockTableDeprecationMessage)
-		config["dynamodb_table"] = config["lock_table"]
-		delete(config, "lock_table")
-	}
-
-	// Delete custom S3 and DynamoDB tags that are only used in Terragrunt config and not in Terraform's backend
-	for _, key := range terragruntOnlyConfigs {
-		delete(config, key)
```

Comment:
That does seem like a bad side effect to have in `configValuesEqual`, but as a sanity check, are we removing or ignoring these Terragrunt-specific configs somewhere else now? That is, we shouldn't be calling `terraform init -backend-config=xxx=yyy` where `xxx` is some Terragrunt-only flag.

Reply:
Yes: https://github.com/gruntwork-io/terragrunt/blob/master/remote/remote_state_s3.go#L271

Note that the generator calls this function to get the stripped down config: https://github.com/gruntwork-io/terragrunt/blob/master/remote/remote_state.go#L181

Reply:
Roger, thx for confirming!